Sidebar for Token Status Page

Gh05d opened the pull request.

daltonfury42 commented:
Can you pull upstream new-frontend and fix the conflicts?

```
git remote add upstream https://github.com/SimplQ/simplQ-frontend.git
git checkout ticket-291
git pull upstream new-frontend
```

maaverik commented on simplq/src/styles/sidePanel.module.scss:
```diff
@@ -1,12 +1,17 @@
-  > div {
+  > button {
     padding: 0.5rem 1rem;
-    padding-right: 4rem;
```
This sidePanel is used in the admin page too. This change would break that page. Could you take a look at /Admin/AdminSidePanel and convert its components to buttons from divs the same way? Or you could leave '> div' along with '> button' in case you get stuck with modifying the admin sidepanel so that this change doesn't break the admin page.

Gh05d replied:
I can take a look. By the way, I just submitted a pull request for a ticket for the AdminPanel, but it covers the Header.

daltonfury42 replied:
Let's convert the admin components to button as well.

Gh05d commented:
> ://github.com/SimplQ/simplQ-frontend.git

I just did it, but when I do `git status` now, all the pulled in files are shown as to be committed:

modified:   ../README.md
	modified:   package-lock.json
	modified:   package.json
	new file:   public/images/alarm-24px.svg
	new file:   public/images/expand_more.svg
	new file:   public/images/mobile_friendly-24px.svg
	new file:   public/images/people-24px.svg
	new file:   public/images/share-24px.svg
	new file:   public/images/wand.svg
	modified:   src/components/Routes.jsx
	new file:   src/components/common/ClickableLogo.jsx
	modified:   src/components/common/Footer.jsx
	deleted:    src/components/common/slider/Slider.jsx
	deleted:    src/components/common/slider/Slider.stories.jsx
	deleted:    src/components/common/slider/index.jsx
	modified:   src/components/pages/Admin/ShareQueue.jsx
	modified:   src/components/pages/Admin/index.jsx
	modified:   src/components/pages/Home/CreateJoinForm.jsx
	modified:   src/components/pages/Home/Nav/Burger.jsx
	modified:   src/components/pages/Home/Nav/Navbar.jsx
	modified:   src/components/pages/Home/StaticInfos.jsx
	modified:   src/components/pages/Join/Form.jsx
	modified:   src/components/pages/Join/index.jsx
	modified:   src/styles/buttons.module.scss
	modified:   src/styles/footer.module.scss
	modified:   src/styles/homePage.module.scss
	modified:   src/styles/joinPage.module.scss
	new file:   src/styles/logo.module.scss
	modified:   src/styles/pageNotFound.module.scss

So pushing now would have the same effect as in the other branch.

daltonfury42 commented:
These changes looks legit you can proceed with the merge. But if you want to, you can create a backup branch from your current branch before you merge.

daltonfury42 commented on simplq/src/components/pages/Status/LeaveQueue.jsx:
```diff
@@ -1,16 +1,23 @@
+export default (props) => {
+  const history = useHistory();
+  function handleClick() {
+    props.leaveQueueHandler();
+    history.push(`/`);
```
Check if leaveQueueHandler returns a promise, if yes, can you change to:
```
props.leaveQueueHandler().then(() => history.push(`/`));
```

Gh05d replied:
Strangely, it wasn't, even though it contains a Promise. I could fix this by adding `async` to `onDeleteClick` (the function which is called).

daltonfury42 replied:
inside onDeleteClick, lets add a return to line 59 [here](https://github.com/Gh05d/simplQ-frontend/blob/ticket-291-new-try/simplq/src/components/pages/Status/QueueStatus.jsx#L59)
```
return TokenService.remove().then(...).catch(...);
```

daltonfury42 replied:
or yeah, await on that call and make it async, upto you

daltonfury42 reviewed (approved):
Approved with minor change requested
